app/controllers/katello/api/v2/sync_plans_controller.rb

```diff
@@ -67,12 +64,12 @@ def create
     param_group :sync_plan
     def update
       sync_date = sync_plan_params.try(:[], :sync_date).try(:to_time)
-
       if !sync_date.nil? && !sync_date.is_a?(Time)
         fail _("Date format is incorrect.")
       end
-
-      sync_task(::Actions::Katello::SyncPlan::Update, @sync_plan, sync_plan_params)
+      old_rec_logic = @sync_plan.foreman_tasks_recurring_logic
+      @sync_plan.update_attributes_with_logics!(sync_plan_params) if sync_plan_params
+      old_rec_logic.cancel if old_rec_logic.id != @sync_plan.foreman_tasks_recurring_logic.id
```

Comment:
@jlsherrill : Cancelling old recurring logic in the controller, and doing rest of the logic inside the model works.

Reply:
why can't that go in the model too?

Reply:
my test last night indicated it should work just fine